### ISSUE TITLE
Invalid .tfw handling while loading pointcloud from .tiff file

### ISSUE THREAD

**Issue by Noxitu** [CloudCompare/CloudCompare#623]
## Current Behavior
Loading any tiff file (as cloud) with following .tfw file:

> 1
> 0
> 0
> -1
> 0
> 0

After loading such cloud all x and y coordinates end with .5.

## Expected Behavior
According to wikipedia (https://en.wikipedia.org/wiki/World_file) as well as QGIS behavior points of cloud with such description should be located in whole coordinates.

## Steps to Reproduce (for bugs)
Create any tiff image. Create tfw file with same name and content mentioned earlier. Open tiff file. 

## Context/Comments
Bug alone doesn't look as a big issue, but it becomes really problematic when working with multifile projects.

For example, if you have two clouds - first (red) with tfw:
> 1 0 0 -1 0 0

and second (green) with:
> .5 0 0 -.5 0 0

currently they will be loaded as follows (yellow cross placed in 0,0):
![current](https://user-images.githubusercontent.com/4777470/33270979-2c60b9a8-d386-11e7-81a6-0fc4d0d02594.png)

I would expect to get following result:
![expected](https://user-images.githubusercontent.com/4777470/33270980-2c7c41a0-d386-11e7-9de4-765898dce715.png)

## Your Environment
* CC Version: v2.10.alpha [64-bit]

**Comment by dgirardeau** [comment]
Are you loading the tiff file as a cloud or a texture quad?

In the first case (as a cloud), we use the GetGeoTransform function of GDAL to get the conversion between pixels and 3D coordinates. One thing I know is that GDAL has two modes when interpreting the pixel location: 'Area' or 'Point':

> Indicates whether a pixel value should be assumed to represent a sampling over the region of the pixel or a point sample at the center of the pixel

This might change how the conversion is done.

Moreover, in the particular case of geotiff files, it seems there's a flag that can be set to specify whether one more should be preferred over the other. Otherwise by default it should be 'Area' I believe.

**Comment by Noxitu** [comment]
I am loading tiff as cloud.

Documentation of GetGeoTransform() indicates that it returns transformation to get upper left corner of pixels - and such corners are currently used as points of rendered pointcloud. I would consider this a bug, since it makes more sense to represent such "pixel" as its center point instead of corner point.

The RasterPixelIsArea is default for GeoTIFFs described by TIFF tags, but worldfiles require the 'Point' mode. But this is correctly handled by GDAL.

I believe issue is in https://github.com/CloudCompare/CloudCompare/blob/master/libs/qCC_io/RasterGridFilter.cpp#L197-198 since transformation that should return coordinates of corner is used as coordinate of actual point.

**Comment by dgirardeau** [comment]
Ok, it makes sense... but then should we always shift the cloud of (-0.5, -0.5) pixels? Whatever the mode? Or only in 'Point' mode?

I never use rasters, and I'm a bit lazy this evening :D

**Comment by Noxitu** [comment]
I haven't check nor I have TIFFs with geo tags, but I would trust GDAL documentation and always shift cloud if you use GetGeoTransform() - altough I think it is (0.5, 0.5) shift (pixel (0, 0) 3d coordinates should be calculated as it was (0.5, 0.5)).

Altough I not use loading as quad I noticed unexpected "-0.5" in the code and it seems for me that it is also incorrect.

I've attached zip with what I would consider correct behaviour - "expected_correct.tiff" after loading should produce cloud and quad positioned as data in Correct.bin.

[Issue #623.zip](https://github.com/CloudCompare/CloudCompare/files/1517153/Issue.623.zip)
